packages/xod-arduino/src/transpiler.js

```diff
@@ -562,30 +562,41 @@ const transformProjectWithImpls = def(
     R.compose(
       R.chain(checkForNativePatchesWithTooManyOutputs),
       // :: Either Error TProject
-      R.map(({ transformedProject, nodeIdsMap }) => {
+      R.chain(({ transformedProject, nodeIdsMap }) => {
         const patches = createTPatches(path, transformedProject, project);
-        return R.merge(
-          {
-            config: {
-              XOD_DEBUG: liveness === LIVENESS.DEBUG,
-              XOD_SIMULATION: liveness === LIVENESS.SIMULATION,
-              literals: R.compose(
-                R.map(literal => {
-                  const key = R.tail(literal); // Get rid of leading `=`
-                  return {
-                    key,
-                    value: xodGlobals[key],
-                  };
-                }),
-                R.filter(isAmong(XP.GLOBALS_LITERALS)),
-                listBoundLiterals
-              )(path, transformedProject),
-            },
-          },
-          R.applySpec({
-            patches: R.always(patches),
-            nodes: createTNodes(path, patches, nodeIdsMap),
-          })(transformedProject)
+        const eitherLiterals = R.compose(
+          R.sequence(Either.of),
+          R.map(literal => {
+            const key = R.tail(literal); // Get rid of leading `=`
+
+            if (!xodGlobals[key])
```

Comment:
Better check for key existence. Otherwise, it will be source of bugs when we’d add numbers, booleans, and strings which may be empty